### ISSUE TITLE
flake inputs

### ISSUE THREAD

**Issue by blaggacao** [fluidattacks/makes#540]
It seems currently not documented (nor ergonomic) to fetch flake in `config.inputs`.

I'd propose a new argument: `fetchFlakeInput` which shells out to `flake.lock` to get the relevant info:

```nix
# makes.nix
{
  inputs = {
    cardano-node = fetchFlakeInput "cardano-node";
  };
}
```

**Comment by blaggacao** [comment]
This can be a trampoline to substitute `makes.lock.nix` and other hard coded inputs in the case a `flake.lock` file is detected and the corresponding "magic" input is present.

**Comment by blaggacao** [comment]
Maybe we can rather implement this in the `inputs` option itself so that it gets auto-populated with `flake.lock` inputs if that file is present... :thinking:

**Comment by blaggacao** [comment]
I think I'll iterate on this one, for now.

**Comment by blaggacao** [comment]
closed via #561
